apps/website/src/components/courses/GroupSwitchModal.tsx

```diff
@@ -38,6 +48,56 @@ const formatDiscussionDateTime = (timestamp: number): string => {
   });
 };
 
+const formatTimezone = () => {
+  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+  const now = new Date();
+  const offsetMinutes = now.getTimezoneOffset();
+  const offsetHours = Math.abs(offsetMinutes) / 60;
+  const offsetSign = offsetMinutes <= 0 ? '+' : '-';
+  const offsetFormatted = `${offsetSign}${Math.floor(offsetHours).toString().padStart(2, '0')}:${(Math.abs(offsetMinutes) % 60).toString().padStart(2, '0')}`;
+  const cityName = timezone.split('/').pop()?.replace(/_/g, ' ') || timezone;
+  return `(GMT ${offsetFormatted}) ${cityName}`;
+};
+
+const getGroupSwitchDescription = ({
+  isCurrentGroup = false,
+  isSelected,
+  isTemporarySwitch,
+  selectedUnitNumber,
+}: {
+  isCurrentGroup?: boolean;
+  isSelected: boolean;
+  isTemporarySwitch: boolean;
+  selectedUnitNumber?: string;
+}): React.ReactNode => {
+  if (isTemporarySwitch) {
+    if (isCurrentGroup) {
+      return isSelected
+        ? <span>You are switching out of this discussion</span>
+        : <span className="text-[#0037FF]">You are attending this discussion</span>;
+    }
+
+    if (isSelected && selectedUnitNumber !== undefined) {
+      return <span className="text-[#0037FF]">You are joining this group for unit {selectedUnitNumber}</span>;
+    }
+
+    return undefined;
+  }
+
+  // For permanent switches
+  if (isCurrentGroup) {
+    return isSelected
+      ? <span>You are switching out of this group for all upcoming units</span>
```

Comment:
The designs have this as e.g. "for Units 2-12". I decided not not to add this in this PR because:
- I'm not 100% sure what the behaviour is in edge cases, because the switching logic is handled by Airtable. E.g. I'm not sure what happens if you have missed your group's discussion, then try to switch into another group where it hasn't started yet. I thought it was better to way "all upcoming units" to not promise anything that might not be true
- This PR was getting quite large overall so I wanted to put it up for review and punt some details like this to a later PR